AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/LocalStorage/LocalStorageError.swift

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Errors interfacing with local storage
+enum LocalStorageError: Error {
+    case missingConnection
+    case invalidDatabase(path: String, Error? = nil)
```

Comment:
nit: perhaps we can rename this to `invalidStorage` since this `LocalStorageError` is now decoupled from SQLite?